contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+          Detector, etc,
+
+          It is intended, to include this functionality later to "older" beam
+          components, such as NXsource, NXdetector, NXlens, etc.
+          Until this is possbible, auxillary beam devices have to be created,
+          for each "older" beam component instead, to allow a beam path description.
+          To link the auxillary beam device to the real device properties, the
+          attribute \@device should be used.
+        \@device:
+          exists: recommended
+          doc: |
+            Reference to beam device, which is described by a NeXus concept
+            (e.g. for NXsource, entry/instrument/source_TYPE).
+        previous_device:
+          exists: recommended
+          doc: |
+            Reference to the previous beam device, from which the photon beam came
+            to reach this beam device. A photon source is related as origin by ".".
+            This enables a logical order description of the optical setup.
+      # non-beam path elements
+      sample_stage(NXmanipulator):
+        exists: optional
+        doc: |
+          Sample stage (or manipulator) for positioning of the sample. This should
+          only contain the spatial orientation of movement. 
+        stage_type:
+          exists: optional
+          doc: |
+            Specify the type of the sample stage.
+          enumeration: [manual stage, scanning stage, liquid stage, gas cell, cryostat, heater, other]
+        stage_type_other:
+          exists: optional
+          doc: |
+            If "other" was chosen in stage_type, enter here a free text description
+            of the stage type.
+        reference_frame:
+          exists: optional
+          doc: |
+            This is a placeholder, to describe the relation between the samples
+            coordinate system, laboratory coordinate system and the stage coordinate
+            system. Up to now, use the "beam_sample_relation" to give supporting
+            information.
+        beam_sample_relation:
+          exists: optional
+          doc: |
+            Description of relation of the beam with the sample. How dit the
+            sample hit the beam, e.g. 'center of sample, long edge parallel
+            to the plane of incidence'.
+        device_information(NXfabrication):
+      temperature_sensor(NXsensor):
+        exists: recommended
+        name:
+          exists: recommended
+        measurement:
+          enumeration: [temperature]
+        type:
+          exists: optional
+        value(NX_FLOAT):
+        device_information(NXfabrication):
+      temp_control_TYPE(NXactuator):
+        doc: 
+          Type of control for the sample temperature. Replace TYPE by 
+          "cryostat" or "heater" to specify it. 
+        exists: optional
+        name:
+          exists: recommended
+        physical_quantity:
+          enumeration: [temperature]
+        cryo_or_heater:
```

Comment:
Could this not be some third, unnamed device? This seems quite limiting.

Reply:
Maybe I dont get the point. you may have several types of NXactuator, and either you cool down or heat up? Not sure what is the missing third possible device?

Reply:
I feel the term `cryo` doesn't necessarily cover all cooling devices. How about air cooling? Probably nitpicking though.

Reply:
Ive addressed it with cooler. This is then generic enough I think.